### PR TITLE
DOC: Add a more welcoming tone for new contributors

### LINKED ISSUE #17579: DOC: Revise doc contribution intro to encourage contributors
Update the documentation contributor README to be more welcoming and encouraging.

### PR DESCRIPTION
**Pull request by willingc**
- [x] closes #17579 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -583,6 +583,12 @@ PyPy
 - Fix :func:`DataFrame.memory_usage` to support PyPy. Objects on PyPy do not have a fixed size,
   so an approximation is used instead (:issue:`17228`)
 
+Documentation
+^^^^^^^^^^^^^
+
+- Improved docs for documentation contributors (:issue:`17579`)
+
+
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: Not necessary in this case.  Generally documentation changes don't get a `whatsnew`.

**Comment by gfyoung** [comment]
@willingc : Thanks for the PR!  Reading this, I'm a little confused by which section of the docs is scaring away new contributors.  In fact, if anything, it's scaring away current contributors from writing new docs because it might come across as too technical and less understandable for the general user.

Could you clarify what you found "less friendly" about the existing documentation?

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (willingc) [reply doc/source/whatsnew/v0.21.0.txt]: Why not? They're a contribution, and it demonstrates the project values a documentation contributor. :smile:

Comment (gfyoung) [reply doc/source/whatsnew/v0.21.0.txt]: I think our documentation speaks for itself when it comes to how much we value it. :smile:

That being said, our change-logs are generally for functional changes / bug fixes.

**Comment by willingc** [comment]
> Could you clarify what you found "less friendly" about the existing documentation?

The existing wording reinforces the myth that documentation is not as valuable as code. (i.e. `If you're not the developer type, contributing to the documentation is still of huge value.`) I doubt that the author's intent was to split folks into "developer"/"not developer". It sends the message "even if you can't write code... the docs are *still* valuable".

Just to clarify my thoughts...I think the pandas contributors and documentation are wonderful. Please consider my comments/suggestions as something that I have found in other projects to improve open source documentation and strengthen the contributor community.

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (willingc) [reply doc/source/whatsnew/v0.21.0.txt]: @gfyoung I believe that you value it :smile:

My point is that the cost of putting documentation changes at the end of the what's new are outweighed by the benefits of encouraging contributors to the documentation.

**Comment by gfyoung** [comment]
> The existing wording reinforces the myth that documentation is not as valuable as code

Ah, I see.  So we're definitely trying to not scare away non-devs.  We explicitly refer non-developers because we've had several people file issues who then back away from contributing because they don't believe they're "dev-like" and therefore can't contribute.

That being said, I see no reason with modifying that one sentence (if that's your issue) instead of modifying the entire two paragraphs.  How about this:

"If you're not the developer type, contributing to the documentation is equally as valuable to us."

**Comment by willingc** [comment]
> That being said, I see no reason with modifying that one sentence (if that's your issue) instead of modifying the entire two paragraphs. 

I modified both paragraphs to create an [affirmative tone](https://docs.python.org/devguide/documenting.html#affirmative-tone) and [economy of expression](https://docs.python.org/devguide/documenting.html#economy-of-expression).

I'm happy to just modify the one sentence if that is what you prefer. Just let me know.

**Comment by gfyoung** [comment]
Compared to the original, the modifications dampen the enthusiasm that the author originally intended with these paragraphs.  Therefore, I would prefer to modify that one sentence, and this PR will be on its way.

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (gfyoung) [reply doc/source/whatsnew/v0.21.0.txt]: @willingc : Still would like to be removed.  As I said, we generally don't log documentation changes.

Comment (TomAugspurger) [reply doc/source/whatsnew/v0.21.0.txt]: In the past that’s what we’ve done, but I’m open to logging doc changes here. I could imagine someone reading the change log and seeing that a section has been rewritten and clicking through to see what’s changed.

**Review by TomAugspurger — APPROVED** [review]

**Comment by gfyoung** [comment]
> In the past that’s what we’ve done, but I’m open to logging doc changes here.

Sure, we can certainly discuss that, but perhaps better if we table for a broader discussion, when the original focus was just to improve the contributing docs.

**Comment by TomAugspurger** [comment]
> Sure, we can certainly discuss that, but perhaps better if we table for a broader discussion

This seems like an opportune place to discuss it. @jorisvandenbossche Thoughts?

My thoughts are

Pros:

1. pixels are cheap. Having more items in the whatsnew isn't a bad thing
2. Contributors get to see their work reflected in the release notes
3. Emphasizes that we value doc contributions, they don't have special rules

Cons:

1. The whatsnew is a common source of merge conflicts, which can be tough for new contributors (but we maintainers can handle them in the web UI with little difficulty, so not a big problem)

Overall a 👍  from me

---

@willingc We're aware of the Travis failure (new release of Pyarrow broke a test), so no need to worry about that.

**Comment by jreback** [comment]
this PR is fine. As far as what'snew for documentation, we generally don't do this unless its a newish section, e.g. something we would want to point users to.

**Comment by jorisvandenbossche** [comment]
+1 from me as well. @TomAugspurger gave a good summary.

BTW, about the actual PR. @willingc I find your original rewrite of the first sentence better (which left out the "If you're not the developer type").

**Comment by gfyoung** [comment]
@jorisvandenbossche : I actually liked that part because it directly addresses those who actually don't believe that they are (people say that explicitly as reasons for being wary of contributing).

**Comment by gfyoung** [comment]
> As far as what'snew for documentation, we generally don't do this unless its a newish section, e.g. something we would want to point users to.

@jorisvandenbossche @TomAugspurger : I actually wasn't aware of this, but in light of this comment, is this change really a newish section?

**Comment by TomAugspurger** [comment]
> > As far as what'snew for documentation, we generally don't do this unless its a newish section, e.g. something we would want to point users to.

> @jorisvandenbossche @TomAugspurger : I actually wasn't aware of this, but in light of this comment, is this change really a newish section?

I think we're discussing the merits of changing the informal policy of not having whatsnew notes. Let's not be constrained by past habits :)

**Comment by gfyoung** [comment]
> I think we're discussing the merits of changing the informal policy of not having whatsnew notes.

Right, but I'll bring up my question again: why in this PR?  Something like this is better suited for an issue that the original changes are getting lost in a larger discussion.

In addition, I might add that a decision like this, while not necessarily as heavy as say an API change does have a noticeable impact on what we share in the `whatsnew` documentation.  I'm not getting stuck in old habits, but I am hesitant to alter how we document changes so late in the release cycle (IINM, the end of the month is our goal).

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -389,7 +389,7 @@ New Behavior:
   ---------------------------------------------------------------------------
   ValueError: Of the three parameters: start, end, and periods, exactly two must be specified
 
-  In [3]: pd.period_range(start='2017Q1', end='2017Q4', periods=6, freq='Q')
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: Hmm...what happened here?

Comment (willingc) [reply doc/source/whatsnew/v0.21.0.txt]: Atom stripped the trailing whitespace.

Comment (willingc) [reply doc/source/whatsnew/v0.21.0.txt]: I can add it back if you want.

Comment (gfyoung) [reply doc/source/whatsnew/v0.21.0.txt]: No, make the change by all means!  If this was a Python file, `flake8` would have been unhappy. 😄

**Comment by willingc** [comment]
Ok. I've made one last pass at this PR. I've remove the whatsnew change, but I hope you will reconsider your policy in a future PR. I've put my original first sentence back and made the "you don't need to be a developer" the second sentence. Feel free to make any additional changes by editing the PR directly. Thanks.

**Comment by gfyoung** [comment]
> Ok. I've made one last pass at this PR. I've remove the whatsnew change, but I hope you will reconsider your policy in a future PR.

Absolutely.  I think it's safe to say that you don't really to hope, as we're reconsidering it right now in your PR, for better or worse. 😄

**Comment by gfyoung** [comment]
@willingc : Actually, why don't you open an issue for this (I'm on my phone, so hard to do).  Then we can transfer the conversation there and merge this in.

**Comment by willingc** [comment]
No problem @gfyoung. I'll open the issue. Thanks!

**Review by gfyoung — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks @willingc!

**Pull request merged**